### ISSUE TITLE
CloudFormation Registry resource support - GetAtt rules broken

### ISSUE THREAD

**Issue by carpnick** [aws-cloudformation/cfn-lint#2622]
### CloudFormation Lint Version

cfn-lint 0.73.1

### What operating system are you using?

Mac

### Describe the bug

Currently the `-s` switch brings in the schemas for custom resource extensions.  This allow some cfn-lint rules to work but not all.

As soon as `!GetAtt` rules are evaluated they start breaking.  For example, [E1010](https://github.com/aws-cloudformation/cfn-lint/blob/1d5c4de28c18a6d2a4d5d72f48d28434f1b0bc8e/src/cfnlint/rules/functions/GetAtt.py), breaks with a custom resource.

I have added logging locally and diagnosed it down to [this line](https://github.com/aws-cloudformation/cfn-lint/blob/1d5c4de28c18a6d2a4d5d72f48d28434f1b0bc8e/src/cfnlint/template.py#L225) being the issue causing GetAtt to fail.

Furthermore, the real issue is `cfnlint.helpers.RESOURCE_SPECS` is using [these](https://d1uauaxba7bl26.cloudfront.net/latest/gzip/CloudFormationResourceSpecification.json) to determine which parameters are valid.  

Realistically, if we want to support this in pre-v1, the right answer is to find a way to combine the schemas and `CloudFormationResourceSpecification` as a source of truth for specifications so the rules get the right models.  However in practice this might be difficult.  

@kddejong  asked me to open via discord

### Expected behavior

All cfn-lint rules work with CloudFormation Registry resource custom extensions. 

### Reproduction template

I dont have any public extensions I can share.  So cannot easily provide a YAML here that is usable by someone else.

With that said, this snippit here shows the problem:

```yaml
  IDPGroup:
    Type: xxxxx::AzureAD::Group
    Properties:
      GroupName: "mycustomgroup"
      GroupType: SECURITY
      Owners:
        - OwnerType: USER
          Name: !Ref OwnerEmail
      CredentialAppClientId: !Ref IDPAutoAppClientId
      CredentialAppAPIToken: !Ref IDPAutoAppAPIToken
      CredentialTenantId: !Ref IDPTenantId
  GroupAssignment:
    Type: xxxxx::AzureAD::GroupAssignment
    Properties:
      GroupId: !GetAtt IDPGroup.GroupId
      AppName: !Ref IDPAppName
      CredentialAppClientId: !Ref IDPAutoAppClientId
      CredentialAppAPIToken: !Ref IDPAutoAppAPIToken
      CredentialTenantId: !Ref IDPTenantId
      AppRoleName: "User"

```

**Comment by agupta94** [comment]
Is there any update on this issue? Having similar issues with GetAtt.
